lib/absinthe/phase/document/complexity.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Absinthe.Phase.Document.Complexity do
+  @moduledoc false
+
+  # Analyses document complexity.
+
+  alias Absinthe.{Blueprint, Phase}
+
+  use Absinthe.Phase
+
+  @default_complexity 1
+
+  @doc """
+  Run complexity analysis.
+  """
+  @spec run(Blueprint.t, Keyword.t) :: Phase.result_t
+  def run(input, options \\ []) do
+    %{operations: ops} = result = Blueprint.postwalk(input, &handle_node/1)
+    complexity = sum_complexity(ops)
+    max = Keyword.get(options, :max_complexity, :infinity)
+    if complexity > max do
+      {:error, "complexity is #{complexity}, which is above maximum #{max}"}
```

Comment:
Should we split the phase in two? The first phase to analyse complexity and the second to handle it? I agree that it is opaque at present.